File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
       

        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        return left

        """
        index = 0
        for i in range(len(nums)):
            if nums[i] >= target:
                index = i
                break
        else:
            index = i + 1
        return index
        """

File: 35-search-insert-position/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int mid, left = 0, right = nums.size() - 1, ans = nums.size();
        while (left <= right) {
            mid = (left + right) / 2;
            if (nums[mid] >= target){
                right = mid - 1;
                ans = mid;
            }                
            else
                left = mid + 1;
        }
        return ans;
    }
};